src/h5web/Visualizer.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement } from 'react';
+import type { HDF5Entity } from './providers/models';
+import DatasetVisualizer from './dataset-visualizer/DatasetVisualizer';
+import { isDataset, isGroup } from './providers/utils';
+import NexusVisualizer from './nexus-visualizer/NexusVisualizer';
+import styles from './Visualizer.module.css';
+
+interface Props {
+  entity?: HDF5Entity;
+}
+
+function Visualizer(props: Props): ReactElement {
+  const { entity } = props;
+
+  if (entity && isDataset(entity)) {
+    return <DatasetVisualizer dataset={entity} />;
+  }
+
+  if (
+    entity &&
+    isGroup(entity) &&
+    entity.attributes?.find(({ value }) => value === 'NXdata')
+  ) {
+    return <NexusVisualizer group={entity} />;
```

Comment:
The new `NexusVisualizer`. We may end up moving the `NXdata` detection code inside the visualizer... For now, it can live here.

Reply:
Yes, it is fine for now.

Just an idea: perhaps, we will need stronger typings for `NXData` for future features (dedicated type + corresponding typeguard).